### ISSUE TITLE
How to use postgis images have a different tags with pgo?

### ISSUE THREAD

**Issue by jdenisgiguere** [CrunchyData/postgres-operator#1749]
** Which example are you working with? **

I'm trying to create a new postgis cluster with postgres operator 4.4.0. According to the pgo documentation, this would be done by:

```
pgo create cluster hagiscluster \
  --ccp-image=crunchy-postgres-gis-ha \
  --ccp-image-tag=centos7-12.3-3.0-4.4.0
```

**What is the current behavior?**

The tag of `crunchy-postgres-gis-ha` now contains the postgis version, by example: `centos7-12.3-3.0-4.4.0`

This tag is different then the tag required by others docker images. Il will be something like `centos7-12.3-4.4.0` (without the postgis version).

**What is the expected behavior?**

I'm not sure what would be the exact expected behavior to keep things simple. Something that would probably be simple to implement  is to have `crunchy-postgres-gis-ha-3-0:centos7-12.3-4.4.0` and `crunchy-postgres-gis-ha-2-5:centos7-12.3-4.4.0`.

Something that may be more user-centred would be to add a tag `--postgis <version>` and to have `pgo` manage de correct image name and tag.

**Please tell us about your environment:**

* Operating System: PopOs! 20.04 LTS
* Where is this running ( Local, Cloud Provider): GCE
* Storage being used (NFS, Hostpath, Gluster, etc): GCE
* Container Image Tag: :laughing: 
* PostgreSQL Version: 12.3 + postgis 3.0
* Platform (Docker, Kubernetes, OpenShift): Kubernetes
* Platform Version: v1.17.8-gke.17

If possible please run the following on the kubernetes or OpenShift (oc) commands and provide the result:
 kubectl describe yourPodName
vents:
  Type     Reason                  Age                    From                                                          Message
  ----     ------                  ----                   ----                                                          -------
  Normal   Scheduled               38m                    default-scheduler                                             Successfully assigned pgogeometric/geometric-dev-1-589879c98f-sqkk8 to gke-anagraph-cluster-n1-standard-2-pr-b65f2b49-dd4c
  Normal   SuccessfulAttachVolume  38m                    attachdetach-controller                                       AttachVolume.Attach succeeded for volume "pvc-dc8f78be-5297-4acf-8a5c-00440301ed7f"
  Normal   Pulled                  38m                    kubelet, gke-anagraph-cluster-n1-standard-2-pr-b65f2b49-dd4c  Container image "crunchydata/crunchy-postgres-gis-ha:centos7-12.3-3.0-4.4.0" already present on machine
  Normal   Created                 38m                    kubelet, gke-anagraph-cluster-n1-standard-2-pr-b65f2b49-dd4c  Created container database
  Normal   Started                 38m                    kubelet, gke-anagraph-cluster-n1-standard-2-pr-b65f2b49-dd4c  Started container database
  Warning  Failed                  37m (x2 over 38m)      kubelet, gke-anagraph-cluster-n1-standard-2-pr-b65f2b49-dd4c  Error: ErrImagePull
  Normal   Pulling                 37m (x3 over 38m)      kubelet, gke-anagraph-cluster-n1-standard-2-pr-b65f2b49-dd4c  Pulling image "crunchydata/crunchy-collect:centos7-12.3-3.0-4.4.0"
  Warning  Failed                  37m (x3 over 38m)      kubelet, gke-anagraph-cluster-n1-standard-2-pr-b65f2b49-dd4c  Failed to pull image "crunchydata/crunchy-collect:centos7-12.3-3.0-4.4.0": rpc error: code = Unknown desc = Error response from daemon: manifest for crunchydata/crunchy-collect:centos7-12.3-3.0-4.4.0 not found: manifest unknown: manifest unknown
  Warning  Failed                  8m9s (x139 over 38m)   kubelet, gke-anagraph-cluster-n1-standard-2-pr-b65f2b49-dd4c  Error: ImagePullBackOff
  Normal   BackOff                 2m59s (x162 over 38m)  kubelet, gke-anagraph-cluster-n1-standard-2-pr-b65f2b49-dd4c  Back-off pulling image "crunchydata/crunchy-collect:centos7-12.3-3.0-4.4.0"

**Comment by jkatz** [comment]
The tagging is done to be able to support [upgrading PostGIS](https://postgis.net/workshops/postgis-intro/upgrades.html).

The issue stems from the PostgreSQL sidecars not being able to be deployed, which would be a bug, at least on the surface.

The Operator does support using the `RELATED_IMAGE*` format that is specified by OLM. An example of these related image tags can be found here:

[https://github.com/CrunchyData/postgres-operator/blob/master/installers/olm/postgresoperator.csv.images.yaml](https://github.com/CrunchyData/postgres-operator/blob/master/installers/olm/postgresoperator.csv.images.yaml)

If you add the following environmental variable to the `pgo-config`  file for the `operator` container and then restart the Operator, it should pull in the correct `crunchy-collect` container when you deploy a new PostgreSQL cluster:

```yaml
- name: RELATED_IMAGE_CRUNCHY_COLLECT
  value: crunchydata/crunchy-collect:centos7-12.3-4.4.0
```

**Comment by jose-joye** [comment]
@jkatz 
We have the same issue in 4.4.1
Do you plan to fix it in 4.5.
The fix you proposed work well if we only install the version 12. In case we want to allow both 11 and 12 to be installed, this may be an issue since it is hard coded

Regards,
José

**Comment by jkatz** [comment]
Beyond the above suggestion, is there anything prevent from adding a tag to your container registry that's along the lines of `crunchy-collect:centos7-12.4-3.0-4.4.1` on top of `crunchy-collect:centos7-12.4-4.4.1`?

I'll also note that part of this is mooted in 4.5, as `crunchy-collect` is removed in favor of `crunchy-postgres-exporter`, which just has a `centos7-4.5.0` tag.

**Comment by jose-joye** [comment]
@jkatz
Thanks for the idea related to the container register. 
This will do the job :-)

**Comment by jkatz** [comment]
I think given that there are infinite possibilities with tagging (hence the "tag" attributes in the CRD), I'll get something in the docs to describe this situation. I have been building out a new section of the documentation that includes "troubleshooting" guides, so noting this to include.

**Comment by aprilito1965** [comment]
@jkatz
1) It is still impossible to create a postgis image based cluster as in the exampple from docs (Common pgo Client Tasks -> Create a PostgreSQL Cluster with PostGIS)

pgo create cluster hagiscluster \
  --ccp-image=crunchy-postgres-gis-ha \
  --ccp-image-tag=centos7-12.4-3.0-4.5.0

Pgbager tries to pull image with tag centos7-12.4-3.0-4.5.0 instead of centos7-12.4-4.5.0.

2) Also, did not understand how to add env variables to "pgo-config" configmap (with many json files defined) in temporary solution. Can you attach an example with listing?

Anyway, your crunchy products are great! Looking forward to the updates!

**Comment by jkatz** [comment]
This is fixed and backpatched down to the 4.4.x series.
